File: Be/property-service/src/main/java/com/property/dto/response/InvoiceResponse.java
package com.property.dto.response;

import java.math.BigDecimal;
import java.util.Set;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@Builder
@ToString
public class InvoiceResponse {
    BigDecimal cost;

    BigDecimal discount;

    String discountType;

    Set<MonthlyUsageResponse> monthlyUsageResponses;

    Boolean pendingInvoice;

    Boolean isPaid;

    int month;
    int year;
}

File: Fe/client/src/dto/response/invoiceResponse.ts
export type invoiceType = {
	cost: number;
	discount: number;
	discountType: string;
	monthlyUsageResponses: Array<monthlyUsageResponses>;
	pendingInvoice: boolean;
	isPaid: boolean;
	month: number;
	year: number;
}

export type monthlyUsageResponses = {
	cost: number;
	usage: number;
	costType: string;
	price: number;
	name: string;
	unit: string;
}